certbot/plugins/standalone.py

```diff
@@ -114,39 +112,53 @@ def running(self):
 SUPPORTED_CHALLENGES = [challenges.TLSSNI01, challenges.HTTP01]
 
 
-def supported_challenges_validator(data):
-    """Supported challenges validator for the `argparse`.
+class SupportedChallengesAction(argparse.Action):
+    """Action class for parsing standalone_supported_challenges."""
 
-    It should be passed as `type` argument to `add_argument`.
+    def __call__(self, parser, namespace, values, option_string=None):
```

Comment:
This was the only instance of using `set_by_cli()` inside argument parsing. The other places it's used are trying to detect if a value is set by the user or not outside of argument parsing.

`set_by_cli()` was used originally because the validation here was written as an argument for `type` rather than `action` way before `set_by_cli()` was even written. `set_by_cli()` was a simple one line fix until we hit `argparse`'s legacy behavior here.